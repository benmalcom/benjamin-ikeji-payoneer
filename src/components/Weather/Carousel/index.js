import React, { useRef, useState } from 'react';
import Slider from 'react-slick';
import cx from 'classnames';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
import { v1 as uuid } from 'uuid';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import styles from './Carousel.module.scss';

import WeatherCard from '../Forecast';
import { Box, Button } from '@material-ui/core';
import PropTypes from 'prop-types';
import forecastProptypes from '../../../utils/prop-types/forecast';

const settings = {
  dots: false,
  infinite: true,
  speed: 500,
  slidesToShow: 3,
  slidesToScroll: 1,
  arrows: false,
  responsive: [
    {
      breakpoint: 1024,
      settings: {
        slidesToShow: 3,
      },
    },
    {
      breakpoint: 600,
      settings: {
        slidesToShow: 2,
      },
    },
    {
      breakpoint: 480,
      settings: {
        slidesToShow: 1,
      },
    },
  ],
};

const WeatherCarousel = (props) => {
  const { items, onCardItemClick, selectedDate } = props;
  const [currentSlide, setCurrentSlide] = useState(0);
  const sliderRef = useRef(null);

  const gotoNext = () => {
    sliderRef.current.slickNext();
  };

  const gotoPrevious = () => {
    sliderRef.current.slickPrev();
  };

  settings.afterChange = index => setCurrentSlide(index);

  const showLeftArrow = currentSlide > 0;
  const showRightArrow = !!items[currentSlide + 3];

  return <>
    <Box component='div' className={cx(
      styles['controls-container'],
      { [styles['controls-left']]: showLeftArrow && !showRightArrow },
      { [styles['controls-right']]: showRightArrow && !showLeftArrow },
    )}>
      {showLeftArrow && <Button className={styles.controls} onClick={gotoPrevious}>
        <ArrowBackIcon fontSize='large' />
      </Button>}
      {showRightArrow && <Button className={styles.controls} onClick={gotoNext}>
        <ArrowForwardIcon fontSize='large' />
      </Button>}
    </Box>
    <Slider ref={sliderRef} {...settings}>
      {items.map(item => <WeatherCard
        selectedDate={selectedDate}
        onClick={onCardItemClick}
        weatherData={item}
        key={uuid} />)}
    </Slider>
  </>;
};

WeatherCarousel.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape(forecastProptypes)).isRequired,
  onCardItemClick: PropTypes.func.isRequired,
  selectedDate: PropTypes.string,
};

WeatherCarousel.defaultProps = {
  selectedDate:null
};


export default WeatherCarousel;