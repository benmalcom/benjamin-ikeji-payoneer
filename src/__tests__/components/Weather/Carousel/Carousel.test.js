import React from 'react';
import { render } from '@testing-library/react';
import Carousel from 'components/Weather/Carousel';
import mockWeatherData from 'utils/mock-data/weatherData.json';
import { createTestStore } from 'utils/test/test-utils';
import { Provider } from 'react-redux';

describe('Carousel', () => {
  const getComponent = props => render(<Provider store={createTestStore()}><Carousel {...props} /></Provider>);

  it('renders without crash', async () => {
    const items = mockWeatherData;
    const onCardItemClick = jest.fn();
    const selectedDate = null;
    const component = getComponent({
      items,
      onCardItemClick,
      selectedDate,
    });
    expect(component).toMatchSnapshot();
  });
});