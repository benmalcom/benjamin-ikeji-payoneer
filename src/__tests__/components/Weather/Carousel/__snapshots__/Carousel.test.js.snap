// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders without crash 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-1 controls-container"
      />
      <div
        class="slick-slider slick-initialized"
      >
        
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <div
                    class="MuiCardContent-root"
                  >
                    <img
                      alt="forecast-icon"
                      class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                      src="http://openweathermap.org/img/wn/01n@2x.png"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      Clear
                    </p>
                    <h2
                      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                    >
                      277°C
                    </h2>
                    <p
                      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      24 Apr, 2021
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      Average Temp: 
                      288°C
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <div
                    class="MuiCardContent-root"
                  >
                    <img
                      alt="forecast-icon"
                      class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                      src="http://openweathermap.org/img/wn/01n@2x.png"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      Clear
                    </p>
                    <h2
                      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                    >
                      276°C
                    </h2>
                    <p
                      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      24 Apr, 2021
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      Average Temp: 
                      287°C
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-cloned"
              data-index="2"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <div
                    class="MuiCardContent-root"
                  >
                    <img
                      alt="forecast-icon"
                      class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                      src="http://openweathermap.org/img/wn/01n@2x.png"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      Clear
                    </p>
                    <h2
                      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                    >
                      277°C
                    </h2>
                    <p
                      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      24 Apr, 2021
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      Average Temp: 
                      288°C
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="3"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <div
                    class="MuiCardContent-root"
                  >
                    <img
                      alt="forecast-icon"
                      class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                      src="http://openweathermap.org/img/wn/01n@2x.png"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      Clear
                    </p>
                    <h2
                      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                    >
                      276°C
                    </h2>
                    <p
                      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      24 Apr, 2021
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      Average Temp: 
                      287°C
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-1 controls-container"
    />
    <div
      class="slick-slider slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  class="MuiCardContent-root"
                >
                  <img
                    alt="forecast-icon"
                    class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                    src="http://openweathermap.org/img/wn/01n@2x.png"
                  />
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    Clear
                  </p>
                  <h2
                    class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                  >
                    277°C
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    24 Apr, 2021
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    Average Temp: 
                    288°C
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  class="MuiCardContent-root"
                >
                  <img
                    alt="forecast-icon"
                    class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                    src="http://openweathermap.org/img/wn/01n@2x.png"
                  />
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    Clear
                  </p>
                  <h2
                    class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                  >
                    276°C
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    24 Apr, 2021
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    Average Temp: 
                    287°C
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-cloned"
            data-index="2"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  class="MuiCardContent-root"
                >
                  <img
                    alt="forecast-icon"
                    class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                    src="http://openweathermap.org/img/wn/01n@2x.png"
                  />
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    Clear
                  </p>
                  <h2
                    class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                  >
                    277°C
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    24 Apr, 2021
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    Average Temp: 
                    288°C
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="3"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  class="MuiCardContent-root"
                >
                  <img
                    alt="forecast-icon"
                    class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                    src="http://openweathermap.org/img/wn/01n@2x.png"
                  />
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    Clear
                  </p>
                  <h2
                    class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                  >
                    276°C
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    24 Apr, 2021
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                  >
                    Average Temp: 
                    287°C
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
